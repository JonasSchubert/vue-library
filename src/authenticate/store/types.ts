export const ActionTypes = {
  login: 'login',
  logout: 'logout',
  validateToken: 'validate-token'
};

export const GetterTypes = {
  error: 'error',
  isAdministrator: 'isAdministrator',
  isLoading: 'isLoading',
  isUser: 'isUser',
  loggedIn: 'loggedIn',
  token: 'token',
  userRoles: 'userRoles'
};

export const ModuleType = 'authenticate';

export const MutationTypes = {
  setError: 'setError',
  setIsLoading: 'setIsLoading',
  setToken: 'setToken',
  setUserRoles: 'setUserRoles'
};

export const RouteTypes = {
  authenticate: 'authenticate'
};
