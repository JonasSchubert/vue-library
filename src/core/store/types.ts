export const GetterTypes = {
  isAnythingLoading: 'isAnythingLoading'
};

export const MutationTypes = {
  decreaseRequests: 'decreaseRequests',
  increaseRequests: 'increaseRequests'
};
