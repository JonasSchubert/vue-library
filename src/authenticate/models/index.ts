export * from './authenticate-module-setup.d';
export * from './authenticate-state.d';
export * from './base-authenticate-state.d';
export * from './login-data.d';
export * from './login-response.d';
export * from './validate-token-data.d';
