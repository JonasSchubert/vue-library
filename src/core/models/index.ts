export * from './base-state.d';
export * from './entity.d';
