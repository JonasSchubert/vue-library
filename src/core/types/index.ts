export * from './ga-colors.type';
