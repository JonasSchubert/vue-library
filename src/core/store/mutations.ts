import { MutationTree } from 'vuex';
import { BaseState } from '../models';
import { MutationTypes } from './types';

export const createMutations = <T extends BaseState>(): MutationTree<T> => ({
  [MutationTypes.decreaseRequests]: (state: T): void => { state.requests -= 1; },
  [MutationTypes.increaseRequests]: (state: T): void => { state.requests += 1; }
});
