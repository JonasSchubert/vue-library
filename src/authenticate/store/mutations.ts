import { MutationTree } from 'vuex';
import { MutationTypes } from './types';
import { AuthenticateModuleSetup, AuthenticateState, LoginResponse } from '../models';
import { writeCookie, deleteCookie } from '../../core/cookie';

export const createMutations = <T extends LoginResponse>({ cookieKeyAuthenticationToken, daysTilExpiredAuthenticationCookie }: AuthenticateModuleSetup): MutationTree<AuthenticateState<T>> => ({
  [MutationTypes.setData]: (state: AuthenticateState<T>, { data, saveLoginDataTemporary }: { data: T | undefined; saveLoginDataTemporary: boolean }): void => {
    state.data = data;

    if (data?.token !== undefined && saveLoginDataTemporary) {
      writeCookie(cookieKeyAuthenticationToken, data.token, daysTilExpiredAuthenticationCookie);
    } else if (!!data && data.token === undefined) {
      deleteCookie(cookieKeyAuthenticationToken);
    }
  },
  [MutationTypes.setError]: (state: AuthenticateState<T>, { error }: { error: Error | undefined }): void => {
    state.error = error;
  },
  [MutationTypes.setIsLoading]: (state: AuthenticateState<T>, { isLoading }: { isLoading: boolean }): void => {
    state.isLoading = isLoading;
  }
});
