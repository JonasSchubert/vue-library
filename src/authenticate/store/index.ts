import { Module } from 'vuex';
import { AuthenticateModuleSetup, AuthenticateState, LoginResponse } from '../models';
import { createActions } from './actions';
import { createGetters } from './getters';
import { createMutations } from './mutations';
import { createAuthenticateState } from './state';

export const createAuthenticateModule = <TRootState, T extends LoginResponse>({ cookieKeyAuthenticationToken, customGetters, daysTilExpiredAuthenticationCookie }: AuthenticateModuleSetup<TRootState, T>): Module<AuthenticateState<T>, TRootState> => ({
  namespaced: true,
  actions: createActions(),
  getters: createGetters(customGetters),
  mutations: createMutations({ cookieKeyAuthenticationToken, daysTilExpiredAuthenticationCookie }),
  state: createAuthenticateState()
});
