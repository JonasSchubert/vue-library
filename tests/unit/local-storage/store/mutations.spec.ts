import { MutationTree } from 'vuex';
import { LocalStorageHistory, LocalStorageSettings, LocalStorageState } from '@/local-storage/models';
import { createMutations } from '@/local-storage/store/mutations';
import { MutationTypes } from '@/local-storage/store/types';

describe('local-storage/mutations', () => {
  let state: LocalStorageState;

  const mutations: MutationTree<LocalStorageState> = createMutations();

  beforeEach(() => {
    state = {
      history: { lastRoutes: [] },
      settings: { isDarkTheme: false }
    };
  });

  describe(`${MutationTypes.setHistory}`, () => {
    test('should set history', () => {
      // Arrange
      const history: LocalStorageHistory = {
        lastRoutes: []
      };

      // Act
      mutations[MutationTypes.setHistory](state, { history });

      // Assert
      expect(state.history).toMatchSnapshot();
    });
  });

  describe(`${MutationTypes.setSettings}`, () => {
    test('should set settings', () => {
      // Arrange
      const settings: LocalStorageSettings = {
        isDarkTheme: true
      };

      // Act
      mutations[MutationTypes.setSettings](state, { settings });

      // Assert
      expect(state.settings).toMatchSnapshot();
    });
  });

  describe(`${MutationTypes.updateIsDarkTheme}`, () => {
    test('should update isDarkTheme', () => {
      // Arrange
      const isDarkTheme = true;

      // Act
      mutations[MutationTypes.updateIsDarkTheme](state, { isDarkTheme });

      // Assert
      expect(state.settings).toMatchSnapshot();
    });
  });

  describe(`${MutationTypes.updateLastRoutes}`, () => {
    test('should update last routes', () => {
      // Arrange
      const lastRoutes: any[] = [];

      // Act
      mutations[MutationTypes.updateLastRoutes](state, { lastRoutes });

      // Assert
      expect(state.history).toMatchSnapshot();
    });
  });
});
