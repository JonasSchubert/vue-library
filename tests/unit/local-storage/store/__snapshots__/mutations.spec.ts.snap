// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`local-storage/mutations setHistory should set history 1`] = `
Object {
  "lastRoutes": Array [],
}
`;

exports[`local-storage/mutations setSettings should set settings 1`] = `
Object {
  "isDarkTheme": true,
}
`;

exports[`local-storage/mutations updateIsDarkTheme should update isDarkTheme 1`] = `
Object {
  "isDarkTheme": true,
}
`;

exports[`local-storage/mutations updateLastRoutes should update last routes 1`] = `
Object {
  "lastRoutes": Array [],
}
`;
